File: solutions/13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> roman = new HashMap<>();
        roman.put('I', 1);
        roman.put('V', 5);
        roman.put('X', 10);
        roman.put('L', 50);
        roman.put('C', 100);
        roman.put('D', 500);
        roman.put('M', 1000);
        
        int n = s.length();
        int result = roman.get(s.charAt(n-1));
        
        for(int i = n-2; i >= 0; i--) {
            if(roman.get(s.charAt(i)) < roman.get(s.charAt(i+1))) {
                result -= roman.get(s.charAt(i));
            } else {
                result += roman.get(s.charAt(i));
            }
        }
        
        return result;
    }
}

File: solutions/13-roman-to-integer/roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        
        roman = {"I": 1, "V":5, "X":10, "L":50, "C":100, "D":500, "M":1000}
        
        n = len(s)
        result = roman[s[n-1]]
        
        for i in range(n-2, -1, -1):
            
            if roman[s[i]] < roman[s[i+1]]:
                result = result - roman[s[i]]
            else:
                result = result + roman[s[i]]
        
        return result
        